### DISCUSSION TITLE
Interpreting the SHAP values presented in layered violin plot in SHAP-library for Scikit binary RandomForestClassifier

### DISCUSSION THREAD

**Discussion by lukepohjankukka** [shap/shap#3683] [Q&A]
I am using the SHAP-library for computing feature Shapley values for a binary [RandomForestClassifier ](https://scikit-learn.org/0.15/modules/generated/sklearn.ensemble.RandomForestClassifier.html) which has naturally two outputs, 0 or 1. The forest itself consists from 100 decision tree classifiers.

I used (I assume selection happens automatically on SHAP-library based on the model) the [TreeExplainer](https://shap.readthedocs.io/en/latest/generated/shap.TreeExplainer.html#shap.TreeExplainer) to calculate the shap values ([shap_values](https://shap.readthedocs.io/en/latest/generated/shap.TreeExplainer.html#shap.TreeExplainer.shap_values)-function) and used the [layered violin plot](https://shap.readthedocs.io/en/latest/example_notebooks/api_examples/plots/violin.html#The-layered-Violin-Summary-Plot) to visualize them. I use mostly all default parameters in SHAP-library functions (with the exception of plotting layered violin plot).

The layered violin plot has on the x-axis (see the link) _SHAP value (impact on model output)_ and on my plot, the SHAP values on the plot range from -0.4 to 0.4 with 0 on the center of the violin plot. I've been trying to clarify to myself **what this SHAP value actually means in my case of a random forest binary classifier?** Note, in my case, **I am considering the SHAP value for the positive class (1)**, since the shap_value-function returns to me a np.ndarray of size `(number_of_samples, number_of_features, 2)`, call it `shap_matrix`, and for the plot, I give the array: `shap_matrix[:, :, 1]`, that is, the shap values for the positive class.

So my question is: what does this SHAP value on the layered violin plot mean for a Scikit-learn RandomForestClassifier with binary outputs 0 or 1? Is it the log-odds? or is it the the change in the model output? But if it's the change in model output, then what does "change in model output of e.g. -0.1 or 0.2" mean in the case where the model output is either 0 or 1?
